Ensure `support_post_types` option returns an array

pfefferle opened the pull request:
Adds a filter to guarantee the 'activitypub_support_post_types' option always returns an array, preventing potential type errors when the option is empty or not set.

<!--- Provide a general summary of your changes in the Title above -->

Fixes #2332

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Adds a new filter callback support_post_types_ensure_array to validate the option value

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to '..'
*

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [X] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [X] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [X] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Added a safeguard to ensure the plugin works correctly even when no post types are selected.

</details>

pfefferle committed:
Ensure support_post_types option returns an array

Adds a filter to guarantee the 'activitypub_support_post_types' option always returns an array, preventing potential type errors when the option is empty or not set.

matticbot committed:
Add changelog

obenland commented:
For the settings field [we cast the option to an array](https://github.com/Automattic/wordpress-activitypub/blob/trunk/includes/wp-admin/class-settings-fields.php#L287), could we do the same in `Post_Types` to fix that warning?

pfefferle commented:
You mean, simply cast the return instead of returning an array when empty? Sure!

pfefferle committed:
Simplify support_post_types_ensure_array logic

Refactored support_post_types_ensure_array to cast input to array directly, removing redundant empty and type checks.

pfefferle committed:
Merge branch 'trunk' into fix/2332

obenland commented:
Apologies, I should have been clearer.

I meant, could we cast the option call to an array [where the warning gets thrown in `Post_Types`](https://github.com/Automattic/wordpress-activitypub/blob/trunk/includes/class-post-types.php#L42), like we do in settings fields?

```diff
- foreach ( \get_option( 'activitypub_support_post_types', array( 'post' ) ) as $post_type ) {
+ foreach ( (array) \get_option( 'activitypub_support_post_types', array( 'post' ) ) as $post_type ) {
```

pfefferle commented:
The problem is, that this is not the only place! And having to add that to every call seemed too fragile to me: https://github.com/search?q=repo%3AAutomattic%2Fwordpress-activitypub%20get_option(%20%27activitypub_support_post_types%27&type=code

We already missed to add it to every call!

pfefferle commented:
But happy to add it to every call instead...

pfefferle committed:
Merge branch 'trunk' into fix/2332

obenland commented:
That's helpful, thanks! Want to remove the unnecessary type cast from settings field and the rest cache integration?

pfefferle committed:
Remove unnecessary array cast for post types option

Eliminated redundant (array) casting when retrieving 'activitypub_support_post_types' option in both settings fields and REST cache classes. This streamlines the code as the default value is already an array.
